### ISSUE TITLE
[bug] `no-missing-component-display-name` not work with early return component

### ISSUE THREAD

**Issue by hyoban** [Rel1cx/eslint-react#177]
### Describe the bug

The following code should not report an error

```tsx
import { forwardRef, ReactNode } from 'react';

interface Props {
  children?: ReactNode;
}
export type Ref = HTMLButtonElement;

const FancyButton = forwardRef<Ref, Props>((props, ref) => {
  // It's works without this
  if (!props.children) return null;
  return (
    <button ref={ref} className="MyClassName" type="button">
      {props.children}
      hell
    </button>
  );
});

FancyButton.displayName = 'FancyButton';

export default function App() {
  return <FancyButton />;
}
```

```
/home/projects/vitejs-vite-bngjjk/src/app.tsx
  8:44  warning  no missing display name  @eslint-react/react/no-missing-component-display-name

✖ 1 problem (0 errors, 1 warning)
```

### Reproduction

https://stackblitz.com/edit/vitejs-vite-bngjjk?file=src%2Fapp.tsx

### Platform and versions

```Text
In stackblitz with "@eslint-react/eslint-plugin": "^0.8.9"
```

**Comment by Rel1cx** [comment]
Thanks for the reproduction, I'll add it to the test cases to prevent it from coming up again.
